src/test/java/com/connexta/ingest/IngestITests.java

```diff
@@ -107,29 +174,82 @@ public void testSuccessfulIngestRequest() throws Exception {
                         .put("message", "The ID asdf has been accepted")
                         .toString()));
 
-    mvc.perform(
-            multipart("/ingest")
-                .file("file", TEST_FILE)
-                .file("metacard", TEST_METACARD)
-                .param("correlationId", "000f4e4a")
-                .header("Accept-Version", "1.2.1")
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.MULTIPART_FORM_DATA))
-        .andExpect(status().isAccepted());
+    final MultipartBodyBuilder builder = new MultipartBodyBuilder();
+    builder.part(
+        "file",
+        new ByteArrayResource(TEST_FILE) {
+
+          @Override
+          public long contentLength() {
+            return TEST_FILE.length;
+          }
+
+          @Override
+          public String getFilename() {
+            return "originalFilename.txt";
+          }
+        });
+    builder.part(
+        "metacard",
+        new ByteArrayResource(TEST_METACARD) {
+          @Override
+          public long contentLength() {
+            return TEST_METACARD.length;
+          }
+
+          @Override
+          public String getFilename() {
+            return "idk why this is needed";
+          }
+        });
+    builder.part("correlationId", "000f4e4a");
+    final MultiValueMap<String, HttpEntity<?>> multipartBody = builder.build();
+    // TODO inject this like we do for the transformApiVersion in TransformClient
```

Comment:
:pencil2: Lines 171-207 break the flow of the method. Move construction of the MultiValueMap to its own method.

Reply:
This comment is outdated.